### PR TITLE
[Cleanup] Remove unused methods in data_utils: 9/N

### PR DESCRIPTION
**Pull request by suchenzang**
Removed:
* infer_language_pair
* filter_by_size
* filter_paired_dataset_indices_by_size
* compute_mask_indices
* get_mem_usage
* lengths_to_padding_mask
* lengths_to_mask

### CODE REVIEW

**Comment by suchenzang** [comment]
Rebased onto main to test again.

File: metaseq/data/data_utils.py [hunk 1]
```diff
@@ -375,154 +273,6 @@ def post_process(sentence: str, symbol: str):
     return sentence
 
 
-def compute_mask_indices(
```
Comment (stephenroller) [inline metaseq/data/data_utils.py:378]: Can you double check armen isn’t using this in his branch? It looks like what would normally power bart and therefore cm3

**Review by stephenroller — APPROVED** [review]
Conditional approval

**Comment by suchenzang** [comment]
Checked the CM3 branch for usage of the above:
```
(base) √ metaseq % ag infer_language_pair
metaseq/data/data_utils.py
27:def infer_language_pair(path):

(base) √ metaseq % ag filter_by_size
metaseq/data/base_dataset.py
185:                indices, ignored = data_utils._filter_by_size_dynamic(
189:            indices, ignored = data_utils._filter_by_size_dynamic(

metaseq/data/data_utils.py
154:def _filter_by_size_dynamic(indices, size_fn, max_positions, raise_exception=False):
187:def filter_by_size(indices, dataset, max_positions, raise_exception=False):
201:        "data_utils.filter_by_size is deprecated. "
217:            indices, ignored = _filter_by_size_dynamic(
221:        indices, ignored = _filter_by_size_dynamic(indices, dataset.size, max_positions)

(base) √ metaseq % ag filter_paired_dataset_indices_by_size
metaseq/data/data_utils.py
240:def filter_paired_dataset_indices_by_size(src_sizes, tgt_sizes, indices, max_sizes):

(base) √ metaseq % ag compute_mask_indices
metaseq/data/data_utils.py
378:def compute_mask_indices(

(base) √ metaseq % ag get_mem_usage
metaseq/data/data_utils.py
505:def get_mem_usage():

(base) √ metaseq % ag lengths_to_padding_mask
metaseq/data/data_utils.py
515:def lengths_to_padding_mask(lens: torch.LongTensor) -> torch.BoolTensor:
523:    return ~lengths_to_padding_mask(lens)

(base) √ metaseq % ag lengths_to_mask
metaseq/data/data_utils.py
522:def lengths_to_mask(lens: torch.LongTensor) -> torch.BoolTensor:

(base) √ metaseq % git status
On branch cm3
Your branch is up to date with 'origin/cm3'.

nothing to commit, working tree clean

(base) √ metaseq % git log
commit 24a601605c2a0b435582e5c40b36dd81c28692c8 (HEAD -> cm3, origin/cm3)
Merge: 0f53b8c c9c817d
Author: Armen Aghajanyan <armen.ag@live.com>
Date:   Wed Aug 3 17:55:39 2022 +0000

    Merge remote-tracking branch 'origin/main' into cm3
```
cc @ArmenAg for viz

**Comment by suchenzang** [comment]
cc @lilisierrayu FYI as well on this PR; don't think any of these methods are used in CM3 but a heads up, just in case

**Pull request merged**